--- skipstone-0.9.3/plugins/Zoomer/Makefile.in.orig	2004-06-28 19:28:57.000000000 +0200
+++ skipstone-0.9.3/plugins/Zoomer/Makefile.in	2004-07-01 21:52:19.070662815 +0200
@@ -3,6 +3,7 @@
 include ../../config.mk
 
 CFLAGS += -fPIC
+CXXFLAGS += -fPIC
 LIBS = $(GTK_LIBS) $(MOZILLA_LIBS)
 LDFLAGS = $(PLUGIN_LDFLAGS)
 
--- skipstone-0.9.3/plugins/FavIcon/Makefile.in.orig	2004-06-28 19:28:57.000000000 +0200
+++ skipstone-0.9.3/plugins/FavIcon/Makefile.in	2004-07-01 21:52:11.621453240 +0200
@@ -4,6 +4,7 @@
 
 
 CFLAGS += -fPIC
+CXXFLAGS += -fPIC
 LIBS = $(GTK_LIBS) 
 LDFLAGS = $(PLUGIN_LDFLAGS)
 OBJS = FavIcon.o FavIconMoz.o
