--- skipstone-0.8.3/config.mk.in.orig	Tue Jun 25 12:35:43 2002
+++ skipstone-0.8.3/config.mk.in	Wed Jun 25 15:47:34 2003
@@ -36,6 +36,7 @@
 	-I$(MOZILLA_INCLUDE_HOME)/webbrwsr \
 	-I$(MOZILLA_INCLUDE_HOME)/string \
 	-I$(MOZILLA_INCLUDE_HOME)/xpcom \
+	-I$(MOZILLA_INCLUDE_HOME)/xpcom_obsolete \
 	-I$(MOZILLA_INCLUDE_HOME)/pref \
 	-I$(MOZILLA_INCLUDE_HOME)/gtkembedmoz \
 	-I$(MOZILLA_INCLUDE_HOME)/nspr \
--- skipstone-0.8.3/src/mozilla.cpp.orig	Wed Jun 25 15:44:06 2003
+++ skipstone-0.8.3/src/mozilla.cpp	Wed Jun 25 15:51:48 2003
@@ -281,7 +281,8 @@
 	        nsAutoString s; s.AssignWithConversion(FullPath);
 	        NS_NewLocalFile(s, PR_TRUE, getter_AddRefs(file));
 	     	if (!file) D_RETURN;
-       		persist->SaveURI(linkURI, nsnull, file);
+//     		persist->SaveURI(linkURI, nsnull, file);
+       		persist->SaveURI(linkURI, nsnull, nsnull, nsnull, nsnull, file);
 	}
 	D_RETURN;
 }
