--- skipstone-0.8.3/config.mk.in.orig	Wed Jun 25 15:47:34 2003
+++ skipstone-0.8.3/config.mk.in	Wed Jun 25 15:55:04 2003
@@ -45,7 +45,7 @@
 
 MOZILLA_LIBS = \
 	-L$(MOZILLA_LIB_HOME) \
-	-lgtkembedmoz -lplds4 -lplc4 -lnspr4 -lgtksuperwin -lxpcom
+	-lgtkembedmoz -lplds4 -lplc4 -lnspr4 -lxpcom
 
 CPPFLAGS += -I. -I../ $(GTK_INCLUDES) $(MOZILLA_INCLUDES) 
 
--- skipstone-0.8.3/configure.in.orig	Wed Jun 25 15:44:06 2003
+++ skipstone-0.8.3/configure.in	Wed Jun 25 15:55:49 2003
@@ -97,10 +97,10 @@
 
 # Check GTK
 # Should use GTK m4 macros
-gtk-config --version 1>/dev/null 2>&1
+pkg-config --version 1>/dev/null 2>&1
 if test $? = 0 ; then
-	GTK_LIBS=`gtk-config --libs`
-	GTK_INCLUDES=`gtk-config --cflags`
+	GTK_LIBS=`pkg-config --libs gtk+-2.0`
+	GTK_INCLUDES=`pkg-config --cflags gtk+-2.0`
 else
 	AC_MSG_ERROR(Cannot find gtk-config, please ensure it is in the path)
 fi
--- skipstone-0.8.3/src/interface.c.orig	Tue Jun 25 12:35:42 2002
+++ skipstone-0.8.3/src/interface.c	Wed Jun 25 16:04:16 2003
@@ -385,9 +385,11 @@
      	skipstone->menubar = gtk_menu_bar_new();
      	skipstone->handlebox = gtk_handle_box_new();
 
-	skipstone->bookmark_toolbar = gtk_toolbar_new(GTK_ORIENTATION_HORIZONTAL,GTK_TOOLBAR_TEXT);
-     	gtk_toolbar_set_space_size(GTK_TOOLBAR(skipstone->bookmark_toolbar),0);
-     	gtk_toolbar_set_button_relief(GTK_TOOLBAR(skipstone->bookmark_toolbar), GTK_RELIEF_NONE);
+	skipstone->bookmark_toolbar = gtk_toolbar_new();
+	gtk_toolbar_set_orientation(GTK_TOOLBAR(skipstone->bookmark_toolbar),GTK_ORIENTATION_HORIZONTAL);
+	gtk_toolbar_set_style(GTK_TOOLBAR(skipstone->bookmark_toolbar),GTK_TOOLBAR_TEXT);
+/*   	gtk_toolbar_set_space_size(GTK_TOOLBAR(skipstone->bookmark_toolbar),0);
+     	gtk_toolbar_set_button_relief(GTK_TOOLBAR(skipstone->bookmark_toolbar), GTK_RELIEF_NONE);*/
      	gtk_container_set_border_width(GTK_CONTAINER(skipstone->bookmark_toolbar),2);
      	
      	skipstone->bmarkhandlebox = gtk_handle_box_new();
@@ -529,10 +531,12 @@
 			
 	}
 	     
-     	skipstone->toolbar = gtk_toolbar_new(GTK_ORIENTATION_HORIZONTAL,GTK_TOOLBAR_BOTH);
-     	gtk_toolbar_set_button_relief(GTK_TOOLBAR(skipstone->toolbar), GTK_RELIEF_NONE);
+     	skipstone->toolbar = gtk_toolbar_new();
+	gtk_toolbar_set_orientation(GTK_TOOLBAR(skipstone->toolbar),GTK_ORIENTATION_HORIZONTAL);
+	gtk_toolbar_set_style(GTK_TOOLBAR(skipstone->toolbar),GTK_TOOLBAR_BOTH);
+/*   	gtk_toolbar_set_button_relief(GTK_TOOLBAR(skipstone->toolbar), GTK_RELIEF_NONE);*/
      	gtk_container_set_border_width(GTK_CONTAINER(skipstone->toolbar),2);
-     	gtk_toolbar_set_space_size(GTK_TOOLBAR(skipstone->toolbar),0);
+/*   	gtk_toolbar_set_space_size(GTK_TOOLBAR(skipstone->toolbar),0);*/
      	gtk_container_add(GTK_CONTAINER(skipstone->handlebox), skipstone->toolbar);
 
      	skipstone_plugin_check(skipstone, PLUGIN_IN_TOOLBAR_START);
--- skipstone-0.8.3/src/interface-notebook.c.orig	Tue Jun 25 12:35:42 2002
+++ skipstone-0.8.3/src/interface-notebook.c	Wed Jun 25 16:04:33 2003
@@ -291,10 +291,12 @@
      	skipstone->menuhandlebox = gtk_handle_box_new ();
      	skipstone->accelgroup = gtk_accel_group_new ();
      	skipstone->bmarkhandlebox = gtk_handle_box_new();
-     	skipstone->bookmark_toolbar = gtk_toolbar_new(GTK_ORIENTATION_HORIZONTAL,GTK_TOOLBAR_TEXT);
+     	skipstone->bookmark_toolbar = gtk_toolbar_new();
+	gtk_toolbar_set_orientation(GTK_TOOLBAR(skipstone->bookmark_toolbar),GTK_ORIENTATION_HORIZONTAL);
+	gtk_toolbar_set_style(GTK_TOOLBAR(skipstone->bookmark_toolbar),GTK_TOOLBAR_TEXT);
      
-     	gtk_toolbar_set_space_size(GTK_TOOLBAR(skipstone->bookmark_toolbar),5);
-     	gtk_toolbar_set_button_relief(GTK_TOOLBAR(skipstone->bookmark_toolbar), GTK_RELIEF_NONE);
+/*   	gtk_toolbar_set_space_size(GTK_TOOLBAR(skipstone->bookmark_toolbar),5);
+     	gtk_toolbar_set_button_relief(GTK_TOOLBAR(skipstone->bookmark_toolbar), GTK_RELIEF_NONE);*/
      	gtk_container_set_border_width(GTK_CONTAINER(skipstone->bookmark_toolbar),2);          
      	gtk_window_add_accel_group(GTK_WINDOW(skipstone->window), skipstone->accelgroup);
      	
@@ -815,9 +817,11 @@
      	gtk_menu_append(GTK_MENU(prefmenu), cb_navtoolbar);
 
 
-	skipstone->bookmark_toolbar = bmark_toolbar_widget =gtk_toolbar_new(GTK_ORIENTATION_HORIZONTAL,GTK_TOOLBAR_TEXT);
-     	gtk_toolbar_set_space_size(GTK_TOOLBAR(skipstone->bookmark_toolbar),5);
-     	gtk_toolbar_set_button_relief(GTK_TOOLBAR(skipstone->bookmark_toolbar), GTK_RELIEF_NONE);
+	skipstone->bookmark_toolbar = bmark_toolbar_widget =gtk_toolbar_new();
+	gtk_toolbar_set_orientation(GTK_TOOLBAR(skipstone->bookmark_toolbar),GTK_ORIENTATION_HORIZONTAL);
+	gtk_toolbar_set_style(GTK_TOOLBAR(skipstone->bookmark_toolbar),GTK_TOOLBAR_TEXT);
+/*   	gtk_toolbar_set_space_size(GTK_TOOLBAR(skipstone->bookmark_toolbar),5);
+     	gtk_toolbar_set_button_relief(GTK_TOOLBAR(skipstone->bookmark_toolbar), GTK_RELIEF_NONE);*/
      	gtk_container_set_border_width(GTK_CONTAINER(skipstone->bookmark_toolbar),2);
      	
      	skipstone->bmarkhandlebox = gtk_handle_box_new();
--- skipstone-0.8.3/src/SkipStonePrompter.cpp.orig	Tue Jun 25 12:35:42 2002
+++ skipstone-0.8.3/src/SkipStonePrompter.cpp	Wed Jun 25 16:03:16 2003
@@ -269,7 +269,7 @@
 {
 	
      	D_ENTER;
-     	mWindow = gtk_window_new(GTK_WINDOW_DIALOG);
+     	mWindow = gtk_window_new(GTK_WINDOW_POPUP);
      	gtk_window_set_position(GTK_WINDOW(mWindow), GTK_WIN_POS_MOUSE);
      	gtk_window_set_title(GTK_WINDOW(mWindow), _("SkipStone Prompt"));
      	GtkBox *topLevelVBox = GTK_BOX(gtk_vbox_new(FALSE, 3));
@@ -355,7 +355,7 @@
 SkipStonePrompter::CreateAlertPrompter(int aFlags)
 {
 	D_ENTER;
-     	mWindow = gtk_window_new(GTK_WINDOW_DIALOG);
+     	mWindow = gtk_window_new(GTK_WINDOW_POPUP);
      	gtk_window_set_position(GTK_WINDOW(mWindow), GTK_WIN_POS_MOUSE);
      	gtk_window_set_title(GTK_WINDOW(mWindow), _("SkipStone Prompt"));
      	GtkBox *topLevelVBox = GTK_BOX(gtk_vbox_new(FALSE, 3));
--- skipstone-0.8.3/src/callbacks.c.orig	Tue Jun 25 12:35:42 2002
+++ skipstone-0.8.3/src/callbacks.c	Wed Jun 25 16:07:29 2003
@@ -871,7 +871,7 @@
 {
      	GSList *list = NULL, *remove = NULL;
      	D_ENTER;
-	gtk_signal_handlers_destroy(GTK_OBJECT(main_notebook));
+/*	gtk_signal_handlers_destroy(GTK_OBJECT(main_notebook));*/
      	/* first make windows for notebook tabs */
 	for (list = window_count ; list != NULL ; list = list->next) {
     		SkipStone *skipstone = (SkipStone *)list->data;
@@ -898,7 +898,7 @@
 	}
      	g_slist_free(remove);
         /*  disconnect the gtk_main_quit from the notebook main_window */
-     	gtk_signal_handlers_destroy(GTK_OBJECT(main_window));
+/*     	gtk_signal_handlers_destroy(GTK_OBJECT(main_window));*/
      	gtk_widget_destroy(main_window);
 	clear_variables();
      	D_RETURN;
--- skipstone-0.8.3/src/moz_callbacks.c.orig	Tue Jun 25 12:35:42 2002
+++ skipstone-0.8.3/src/moz_callbacks.c	Wed Jun 25 16:08:30 2003
@@ -50,7 +50,7 @@
 {
      	gchar *title;
 	D_ENTER;
-     	gtk_signal_handlers_destroy(GTK_OBJECT(skipstone->window));
+/*     	gtk_signal_handlers_destroy(GTK_OBJECT(skipstone->window));*/
      	title = GTK_WINDOW(skipstone->window)->title;
 
      	if (title)
--- skipstone-0.8.3/src/skipstone-plugin.c.orig	Tue Jun 25 12:35:42 2002
+++ skipstone-0.8.3/src/skipstone-plugin.c	Wed Jun 25 16:09:36 2003
@@ -250,7 +250,7 @@
 {
      D_ENTER;
      g_return_if_fail(skipstone != NULL);
-     gtk_container_queue_resize(GTK_CONTAINER(skipstone->vbox));
+/*   gtk_container_queue_resize(GTK_CONTAINER(skipstone->vbox));*/
      D_RETURN;
 }
 
