diff -Nur skipstone-0.8.3.orig/configure.in skipstone-0.8.3/configure.in
--- skipstone-0.8.3.orig/configure.in	Wed Jun  5 21:42:36 2002
+++ skipstone-0.8.3/configure.in	Thu Jun  6 17:27:52 2002
@@ -177,6 +177,5 @@
                  plugins/SearchToolBar/Makefile
                  plugins/Throbber/Makefile
                  plugins/Zoomer/Makefile
-		 plugins/Up/Makefile
                  src/Makefile
 		 src/skipstone])
