diff -Nur skipstone-0.8.2.orig/configure.in skipstone-0.8.2/configure.in
--- skipstone-0.8.2.orig/configure.in	Thu Apr 11 05:04:16 2002
+++ skipstone-0.8.2/configure.in	Sat Apr 20 04:44:28 2002
@@ -83,14 +83,6 @@
 	fi
 fi
 
-if test -f $mozilla_libs/chrome/comm.jar ; then
-	MOZILLA_LIB_HOME=$mozilla_libs
-	AC_SUBST(MOZILLA_LIB_HOME)
-else
-	AC_MSG_ERROR(Cannot find mozilla libs - please use --with-mozilla-libs=path_to_mozilla_libs and specify a path)
-fi	
-
-
 # Check GTK
 # Should use GTK m4 macros
 gtk-config --version 1>/dev/null 2>&1
