diff -Nur skipstone-0.7.4.orig/config.mk skipstone-0.7.4/config.mk
--- skipstone-0.7.4.orig/config.mk	Fri Jul 13 19:25:24 2001
+++ skipstone-0.7.4/config.mk	Sun Jul 15 20:19:15 2001
@@ -2,16 +2,11 @@
 PLUGINDIR=$(HOME)/.skipstone/plugins
 
 MOZILLA_INCLUDES = \
-	-I/usr/include/mozilla \
-	-I/usr/local/include/mozilla \
-	-I/usr/include/nspr \
-	-I/usr/local/include/nspr \
-	-I/usr/include/mozilla/nspr \
-	-I/usr/local/include/mozilla/nspr
+	-I/usr/X11R6/include/mozilla \
+	-I/usr/X11R6/include/mozilla/nspr
 
 MOZILLA_LIBS = \
-	-L/usr/lib/mozilla \
-	-L/usr/local/lib/mozilla \
+	-L/usr/X11R6/lib/mozilla \
 	-lgtkembedmoz -lplds4 -lplc4 -lnspr4 -lgtksuperwin -lxpcom
 
 GTK_LIBS = $(shell gtk-config --libs)
@@ -21,6 +16,6 @@
 CXX = g++
 
 CPPFLAGS = $(GTK_INCLUDES) $(MOZILLA_INCLUDES)
-CFLAGS = -g -O2 -Wall
+CFLAGS = $(OPT) -Wall
 CXXFLAGS = $(CFLAGS) -fno-rtti
 
