--- skipstone-0.9.3/src/skipstone.in.orig	2004-06-28 19:28:56.000000000 +0200
+++ skipstone-0.9.3/src/skipstone.in	2004-06-30 23:25:42.795028304 +0200
@@ -1,32 +1,15 @@
 #!/bin/sh
 # allow MOZILLA_FIVE_HOME to override defaults/install path
 if test -n "$MOZILLA_FIVE_HOME"; then
-	LD_LIBRARY_PATH=$MOZILLA_FIVE_HOME:$LD_LIBRARY_PATH
+	if test -n "$LD_LIBRARY_PATH"; then
+		LD_LIBRARY_PATH=$MOZILLA_FIVE_HOME:$LD_LIBRARY_PATH
+	else
+		LD_LIBRARY_PATH=$MOZILLA_FIVE_HOME
+	fi
 	export LD_LIBRARY_PATH
-elif [ -f /usr/lib/libgtkembedmoz.so ]; then
-	export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/mozilla
-	export MOZILLA_FIVE_HOME=/usr/lib/mozilla
-elif  [ -f @MOZILLA_LIB_HOME@/libgtkembedmoz.so ]; then
-	export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:@MOZILLA_LIB_HOME@
-	export MOZILLA_FIVE_HOME=@MOZILLA_LIB_HOME@
-elif  [ -f @MOZILLA_LIB_HOME@/libgtkembedmoz.dylib ]; then
-	export DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:@MOZILLA_LIB_HOME@
-	export MOZILLA_FIVE_HOME=@MOZILLA_LIB_HOME@
-elif [ -f /usr/local/lib/mozilla/libgtkembedmoz.so ]; then
-	export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib/mozilla
-	export MOZILLA_FIVE_HOME=/usr/local/lib/mozilla
-elif [ -f /opt/mozilla/libgtkembedmoz.so ]; then
-        export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/mozilla
-        export MOZILLA_FIVE_HOME=/opt/mozilla
-
-else 
-
-echo "Please set your MOZILLA_FIVE_HOME and LD_LIBRARY_PATH manually as I could not find them"
-     exit
-fi
-
-if [ -f ./skipstone-bin ]; then
-	exec ./skipstone-bin $@
 else
-	exec skipstone-bin $@
-fi
+	MOZILLA_FIVE_HOME="@MOZILLA_LIB_HOME@"
+	export MOZILLA_FIVE_HOME
+fi 
+
+exec @bindir@/skipstone-bin $@
